src/App.js

```diff
@@ -1,5 +1,47 @@
+import LottoMachine from "./LottoMachine.js";
+import Lotto from "./Lotto.js";
+import { getInputWithErrorHandling, printMessage } from "./utils/interface.js";
+import BonusNumber from "./BonusNumber.js";
+import { MESSAGE } from "./constants/messages.js";
+
 class App {
-  async run() {}
+  #lottoMachine;
+  #winningLotto;
+  #bonusNumber;
+
+  async run() {
+    await getInputWithErrorHandling({
+      inputMessage: MESSAGE.INPUT_MONEY,
+      handleInputFn: (input) => {
+        const inputNumber = Number(input);
+        this.#lottoMachine = new LottoMachine(inputNumber);
+      },
+    });
+
+    printMessage(this.#lottoMachine.getTicketsString());
+
+    await getInputWithErrorHandling({
+      inputMessage: MESSAGE.INPUT_WINNING_LOTTO,
+      handleInputFn: (input) => {
+        const numbers = input.split(",").map(Number);
+        this.#winningLotto = new Lotto(numbers);
+      },
+    });
+
+    await getInputWithErrorHandling({
+      inputMessage: MESSAGE.INPUT_BONUS_NUMBER,
+      handleInputFn: (input) => {
+        const inputNumber = Number(input);
+        this.#bonusNumber = new BonusNumber(inputNumber, this.#winningLotto);
+      },
+    });
+
+    this.#lottoMachine.countWinningLotto({
+      winningLotto: this.#winningLotto,
+      bonusNumber: this.#bonusNumber,
+    });
+    printMessage(MESSAGE.OUTPUT_RESULT + this.#lottoMachine.getLottoResultString());
```

Comment:
[javascript airbnb에서, 템플릿 리터럴을 권장](https://github.com/airbnb/javascript?tab=readme-ov-file#es6-template-literals)하고 있어서, 이를 사용하면 어떨까요 ?

Reply:
저두 이 부분 고민하긴 했지만, Airbnb Style Guide에서 권장하는 부분은 중간에 다른 문자가 포함되어 있을 때인 것 같더라구요! 여기서는 다른 문자를 추가할 필요없이, 문자열 두개를 이어붙이면 되는 부분이라서 템플릿 리터럴을 사용하지 않았습니다 😁

Reply:
저는 상황에 따라 다르게 쓰는건 일관성 면에서 떨어진다고 생각해요. 이 경우에는 문자열인데 `+`, 이 경우에는 문자열이니까 템플릿 리터럴을 사용하면 일관적이지 않고, 팀원과 함께 하는 프로젝트라면 혼동을 결국 야기하기 될테니깐요.

airbnb javascript에서 보여준 예시에는 중간에 문자열을 삽입하는 것들만 있지만, 위와 같은 것들도 포괄하고 있다고 생각해요.

또한, `+`는 숫자의 더하기라는 것을 일차원적으로 생각하기에 그러한 측면에서 airbnb에서 `+`보다 문자열임을 나타내는 템플릿 리터럴을 권장하는게 아닐까 생각해요.

Reply:
오 그럴수도 있겠네요..! 🧐 

```suggestion
    printMessage(`${MESSAGE.OUTPUT_RESULT}${this.#lottoMachine.getLottoResultString()}`);
```

저는 뭔가 이런 경우에는 템플릿 리터럴보다 `+`연산자가 깔끔해보이고 직관적이라고 생각했어요! 근데 규성님 얘기도 맞을 것 같아요 ㅋㅋㅋ